### ISSUE TITLE
S2 TreeView traps focus in expand/collapse button when no focusable content

### ISSUE THREAD

**Issue by rkantor7** [adobe/react-spectrum#8111]
### Provide a general summary of the issue here

In an S2 TreeView, if an item with children is not selectable and does not have any actions (so has no focusable internal content), pressing tab moves focus to the expand/collapse button inside the item, and then it becomes stuck. Pressing tab refocuses the same button and never moves forward. You can navigate back with shift+tab, but tabbing again will cause focus to become stuck again.

### 🤔 Expected Behavior?

On tab focus should move out of the TreeView to the next focusable item on the page.

### 😯 Current Behavior

On tab focus gets stuck on the expand/collapse parent button.

### 💁 Possible Solution

This only happens when the item is not selectable and does not have actions, so likely it's something to do with how focus for those internal items is handled. 

### 🔦 Context

We are imbedding a TreeView in our page before the main page content. This issue makes it so a user cannot tab onto the actual page and remains stuck

### 🖥️ Steps to Reproduce

It's reproduced in this codesandbox: https://codesandbox.io/p/sandbox/boring-gauss-cm7pl7. It happens any time there is a TreeView that is not selectable and has no actions within the items, and one of the items has children.

### Version

3.40.0

### What browsers are you seeing the problem on?

Chrome

### What operating system are you using?

Mac Sequoia 15.4

### 🧢 Your Company/Team

Unified Shell

**Comment by LFDanLu** [comment]
Looks like it is because the expand button manages to get a tabindex of `0` on it when it shouldn't be tabbable ever. I think we can replace the following line with `excludeFromTabOrder: true` so that it actually applies the proper tabindex of `-1` at all times. Will need to double check the behavior with mobile screen readers to make sure it doesn't inadvertently break anything
